evalml/model_understanding/graphs.py

```diff
@@ -605,221 +612,250 @@ def partial_dependence(
+                    "dependence is supported.",
+                    code=PartialDependenceErrorCode.TOO_MANY_FEATURES,
+                )
+            if not (
+                all([isinstance(x, str) for x in features])
+                or all([isinstance(x, int) for x in features])
+            ):
+                raise PartialDependenceError(
+                    "Features provided must be a tuple entirely of integers or strings, not a mixture of both.",
+                    code=PartialDependenceErrorCode.FEATURES_ARGUMENT_INCORRECT_TYPES,
+                )
+            X_features = (
+                X.ww.iloc[:, list(features)]
+                if isinstance(features[0], int)
+                else X.ww[list(features)]
             )
-        if not (
-            all([isinstance(x, str) for x in features])
-            or all([isinstance(x, int) for x in features])
-        ):
-            raise ValueError(
-                "Features provided must be a tuple entirely of integers or strings, not a mixture of both."
+        else:
+            X_features = (
+                X.ww.iloc[:, [features]]
+                if isinstance(features, int)
+                else X.ww[[features]]
             )
-        X_features = (
-            X.ww.iloc[:, list(features)]
-            if isinstance(features[0], int)
-            else X.ww[list(features)]
-        )
-    else:
-        X_features = (
-            X.ww.iloc[:, [features]] if isinstance(features, int) else X.ww[[features]]
-        )
-    X_unknown = X_features.ww.select("unknown")
-    if len(X_unknown.columns):
-        # We drop the unknown columns in the pipelines, so we cannot calculate partial dependence for these
-        raise ValueError(
-            f"Columns {X_unknown.columns.values} are of type 'Unknown', which cannot be used for partial dependence"
-        )
-
-    X_not_allowed = X_features.ww.select(["URL", "EmailAddress", "NaturalLanguage"])
-    if len(X_not_allowed.columns):
-        # these three logical types aren't allowed for partial dependence
-        types = sorted(set(X_not_allowed.ww.types["Logical Type"].astype(str).tolist()))
-        raise ValueError(
-            f"Columns {X_not_allowed.columns.tolist()} are of types {types}, which cannot be used for partial dependence"
-        )
-
-    X_cats = X_features.ww.select("categorical")
-    if any(is_categorical):
-        max_num_cats = max(X_cats.ww.describe().loc["nunique"])
-        grid_resolution = max([max_num_cats + 1, grid_resolution])
-
-    X_dt = X_features.ww.select("datetime")
-
-    if isinstance(features, (list, tuple)):
-        feature_names = _get_feature_names_from_str_or_col_index(X, features)
-        if any(is_datetime):
-            raise ValueError(
-                "Two-way partial dependence is not supported for datetime columns."
+        X_unknown = X_features.ww.select("unknown")
```

Comment:
Can we combine this with the following check for "URL", "EmailAddress", "NaturalLanguage"? They all have the `INVALID_FEATURE_TYPE` code :)